### PR TITLE
remove all deprecations cncf.kubernetes

### PR DESCRIPTION
**Pull request by romsharon98**
since we now change the provider version to be 10.0.0 we can now remove all the deprecation warnings.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — CHANGES_REQUESTED** [review]
LGTM
needs to update changelog to list the removals

File: providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst [hunk 1]
```diff
@@ -31,6 +31,34 @@ Changelog
 main
 .....
 
+All deprecated classes, parameters and features have been removed from the Amazon provider package.
```
Comment (eladkal) [inline providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst]:
```suggestion
All deprecated classes, parameters and features have been removed from the Kubernetes provider package.
```

File: providers/src/airflow/providers/amazon/aws/operators/eks.py [hunk 2]
```diff
@@ -40,14 +40,9 @@
...
 from airflow.providers.amazon.aws.utils.waiter_with_logging import wait
+from airflow.providers.cncf.kubernetes.operators.pod import KubernetesPodOperator
 from airflow.providers.cncf.kubernetes.utils.pod_manager import OnFinishAction
 
-try:
-    from airflow.providers.cncf.kubernetes.operators.pod import KubernetesPodOperator
-except ImportError:
-    # preserve backward compatibility for older versions of cncf.kubernetes provider
-    from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
-
```
Comment (eladkal) [inline providers/src/airflow/providers/amazon/aws/operators/eks.py]: lets undo this change. It's part of the aws provider and depend on what aws wants to set as min version for cncf provider

Comment (eladkal) [reply providers/src/airflow/providers/amazon/aws/operators/eks.py]: we can add todo: remove this import when min cncf k8s provider is 10.0

Comment (romsharon98) [reply providers/src/airflow/providers/amazon/aws/operators/eks.py]: added a todo comment

File: providers/tests/cncf/kubernetes/test_pod_generator.py [hunk 3]
```diff
@@ -725,10 +642,7 @@ def test_deserialize_non_existent_model_file(self, caplog, tmp_path):
         ),
     )
     def test_pod_name_confirm_to_max_length(self, input):
-        with pytest.warns(
-            AirflowProviderDeprecationWarning, match="Use `add_pod_suffix` in `kubernetes_helper_functions`"
-        ):
-            actual = PodGenerator.make_unique_pod_id(input)
+        actual = add_unique_suffix(name=input, max_len=100)
```
Comment (romsharon98) [inline providers/tests/cncf/kubernetes/test_pod_generator.py]: the default `max_len` in `add_unique_suffix`  is 63 but when looking at `make_unique_pod_id` the default is 100, I fix the test accordingly but wonder if there is other solution

Comment (eladkal) [reply providers/tests/cncf/kubernetes/test_pod_generator.py]: cc @jedcunningham WDYT?

Comment (jedcunningham) [reply providers/tests/cncf/kubernetes/test_pod_generator.py]: I don't see a big problem here, shorter should be fine. I don't view changing pod names as a breaking change.

And yeah, probably better to refactor this test with the new limit.

**Review by eladkal — APPROVED** [review]
